packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -346,35 +437,95 @@ function WorklistDataGrid(props) {
     }
   }, [hasSelectableRows, intl, onColumnSelect]);
 
-  const handleCellSelection = useCallback((cellRowIdColId, cellCoordinates, cellSelectable) => {
+  const handleCellSelectionChange = useCallback((rowId, columnId, cellCoordinates) => {
     if (!hasSelectableRows) {
-      setAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.cell-selection-template' },
-        { row: cellCoordinates.row + 1, column: cellCoordinates.col + 1 }));
+      setAriaLiveMessage(intl.formatMessage({
+        id: rowId ? 'Terra.worklist-data-grid.cell-selection-template' : 'Terra.worklist-data-grid.cell-selection-cleared',
+      }, { row: cellCoordinates.row + 1, column: cellCoordinates.col + 1 }));
+    }
+    setFocusedRowCol(cellCoordinates.row, cellCoordinates.col, true);
+    if ((rowId !== currentSelectedCell?.rowId) || (columnId !== currentSelectedCell?.columnId)) {
+      setCurrentSelectedCell((rowId && columnId) ? { rowId, columnId } : null);
     }
+  }, [currentSelectedCell?.columnId, currentSelectedCell?.rowId, hasSelectableRows, intl, setFocusedRowCol]);
 
-    setFocusedRow(cellCoordinates.row);
-    setFocusedCol(cellCoordinates.col);
-    setCurrentSelectedCell({ rowId: cellRowIdColId.rowId, columnId: cellRowIdColId.columnId });
+  const mapGridCellToDataCell = useCallback((cellGridCoordinates) => (
+    // The grid has an additional row for the heading and
+    // may have an additional column when when selection is active.
+    { row: cellGridCoordinates.row - 1, col: cellGridCoordinates.col + (hasSelectableRows ? -1 : 0) }
+  ), [hasSelectableRows]);
 
-    if (cellSelectable && onCellSelect) {
-      onCellSelect(cellRowIdColId.rowId, cellRowIdColId.columnId);
+  const handleCellSelection = useCallback((selectionDetails) => {
+    if (!selectionDetails?.selectedByKeyboard) {
+      handleFocus.current = false;
     }
-  }, [hasSelectableRows, intl, onCellSelect]);
 
-  const handleRowSelection = useCallback((rowId, rowIndex, selectedCellCoordinates) => {
-    const rowSelectionMessageId = !rows[rowIndex - 1].isSelected ? 'Terra.worklist-data-grid.row-selection-template' : 'Terra.worklist-data-grid.row-selection-cleared-template';
-    const rowLabel = rows[rowIndex - 1].ariaLabel || (rowIndex + 1);
+    const cellGridCoordinates = { row: selectionDetails.rowIndex, col: selectionDetails.columnIndex };
+    // If current cell is selected, do nothing.
+    if (isCellSelected(cellGridCoordinates.row, cellGridCoordinates.col)) {
+      return;
+    }
+    // Determine if the cell selection should proceed.
+    const cellDataCoordinates = mapGridCellToDataCell(cellGridCoordinates);
+    const cell = rows[cellDataCoordinates.row].cells[cellDataCoordinates.col];
+    if ((cell.isSelectable === false) || cell.isMasked) {
```

Comment:
selectionDetails has an isCellSelectable property.  You can use that as was done in the previous logic.

Reply:
Updated